### PR TITLE
Fix parsers

### PR DESCRIPTION
**Pull request by GeorgosXL**

### CODE REVIEW

**Comment by Nyholm** [comment]
Could you rebase please

**Comment by GeorgosXL** [comment]
Done with rebase

File: Parser/RewriteMediaUrl.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    {
+        $media->setSourceUrl($this->rewriteUrl($media->getSourceUrl()));
+    }
+
+    private function rewriteUrl(string $content): string
+    {
+        $remoteUrl = parse_url($this->remoteUrl);
+        $contentUrl = parse_url($content);
+
+        if (!$remoteUrl) {
```
Comment (Nyholm) [inline Parser/RewriteMediaUrl.php]: Why do we check $remoteUrl here?

File: Parser/RewriteMediaUrl.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            return $content;
+        }
+
+        if ($contentUrl['host'] && $remoteUrl['host'] === $contentUrl['host']) {
```
Comment (Nyholm) [inline Parser/RewriteMediaUrl.php]: What does the `if ($contentUrl['host'] &&` check?

File: Service/LocalImageUploader.php [hunk 3]
```diff
@@ -37,6 +37,11 @@ public function uploadImage(string $url): string
 
         $filename = \basename($url);
 
+        // Check if file already exists
+        if (\file_exists(\sprintf('%s/%s', $this->uploadsFolder, $filename))) {
+            return \sprintf('%s/%s', $this->webPrefix, $filename);
```
Comment (Nyholm) [inline Service/LocalImageUploader.php]: It is a lot of code duplication here.

File: Tests/Parser/RewriteMediaUrlTest.php [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    public function testRewrite($inputUrl)
+    {
+        $media = new Media();
+        $media->setSourceUrl(self::TEST_URL);
+        $imageUploader = $this->getMockBuilder(ImageUploaderInterface::class)
+            ->setMethods(['uploadImage'])
+            ->getMock();
+
+        $parser = new RewriteMediaUrl($inputUrl, $imageUploader);
+        $parser->parseMedia($media);
```
Comment (Nyholm) [inline Tests/Parser/RewriteMediaUrlTest.php]: What do we actually test here?

File: Parser/RewriteMediaUrl.php (hunk 1 above)
Comment (GeorgosXL) [reply Parser/RewriteMediaUrl.php]: if parse_url($this->remoteUrl) returns false, which means that the url is malformed, we don't want to continue with an incorrect url. Therefore we return the original media source url (content)

**Pull request merged**